mysql_mimic/intercept.py

```diff
@@ -60,4 +60,4 @@ def expression_to_value(expression: exp.Expression) -> Any:
         return True
     if expression.name == "OFF":
         return False
-    return expression.name
+    raise ValueError(f"Unexpected expression: {expression}")
```

Comment:
We should raise a https://github.com/kelsin/mysql-mimic/blob/main/mysql_mimic/errors.py#L44

Sometimes I'll go and try to find an appropriate error code here: https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html

But other times I give up and just use the default.

In this case, I think MySQL allows query expressions here: https://dev.mysql.com/doc/refman/8.0/en/set-variable.html

So maybe we throw some kind of "not supported yet" error, like this: https://github.com/kelsin/mysql-mimic/blob/main/mysql_mimic/session.py#L521

Reply:
Yeah this makes a lot of sense. 

I did not go to https://dev.mysql.com/doc/refman/8.0/en/set-variable.html and check expression is allowed